framework/src/main/java/org/checkerframework/framework/stub/StubParser.java

```diff
@@ -1418,6 +1419,17 @@ private Object getValueOfExpressionInAnnotation(
             return convert(((IntegerLiteralExpr) expr).asInt(), valueKind);
         } else if (expr instanceof LongLiteralExpr) {
             return convert(((LongLiteralExpr) expr).asLong(), valueKind);
+        } else if (expr instanceof UnaryExpr) {
+            if (((UnaryExpr) expr).getOperator() == UnaryExpr.Operator.MINUS) {
+                Expression subexpr = ((UnaryExpr) expr).getExpression();
+                if (subexpr instanceof IntegerLiteralExpr) {
```

Comment:
Rather than handling just literals here, how about making a recursive call?
That will handle more cases, and will also avoid duplicating code.

Reply:
This method returns an Object, so we would need a case distinction for the type of the result of the recursive call to then multiply by -1 and convert it to the correct result object.
This would give us the additional case for `expr instanceof FieldAccessExpr || expr instanceof NameExpr` case.
So the code wouldn't be much shorter, but handling the additional case would be nice.

Reply:
addressed